Latest/Model/AppLibrary.swift

```diff
@@ -47,20 +47,43 @@ class AppLibrary {
 	@objc func gatherAppBundles() {
 		self.appSearchQuery.disableUpdates()
 		
-		// Run metadata query to gather all apps
-		self.bundles = self.appSearchQuery.results.compactMap { item -> App.Bundle? in
-			guard let metadata = item as? NSMetadataItem,
-				let path = metadata.value(forAttribute: NSMetadataItemPathKey) as? String else { return nil }
-			
-			let url = URL(fileURLWithPath: path)
-			
-			// Only allow apps in the application folder and outside excluded subfolders
-			if !Self.applicationPaths.contains(where: { url.path.hasPrefix($0) }) || Self.excludedSubfolders.contains(where: { url.path.contains($0) }) {
-				return nil
+		let appSearchQueryResults = self.appSearchQuery.results
+		if appSearchQueryResults.isEmpty {
+			// Look for applications in the /Applications folder manually
+			let fileManager = FileManager.default
+			var bundles = [App.Bundle]()
+			do {
+				for applicationPath in Self.applicationPaths {
+					let appPaths = try fileManager.contentsOfDirectory(atPath: applicationPath)
```

Comment:
The current implementation only looks for files within the application paths themselves. However, some people organise their Applications folder with nested folders. I checked the `FileManager` API and there is an enumerator method:
`fileManager.enumerator(at: URL(fileURLWithPath: applicationPath), includingPropertiesForKeys: nil, options: [.skipsHiddenFiles, .skipsPackageDescendants], errorHandler: nil)`

I think we can safely ignore hidden files and also packages (.app files).

Reply:
It actually isn't that simple. I think we should do manual traversing, because this enumerator, obviously, skips all the .app files, and in my case returns nothing.
If we don't ignore packages, we'll be buried in app bundles very-very deep :)
Let's consider adding this as an improvement later on, instead.